runtimes/core/src/lib.rs

```diff
@@ -52,8 +53,61 @@ pub mod encore {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ProcessConfig {
+    hosted_services: Vec<String>,
+    hosted_gateways: Vec<String>,
+    local_service_ports: HashMap<String, u16>,
+}
+
+impl ProcessConfig {
+    pub fn apply(&self, cfg: &mut runtimepb::RuntimeConfig) {
+        let deployment = cfg.deployment.get_or_insert_with(Default::default);
+
+        deployment.hosted_services = self
+            .hosted_services
+            .iter()
+            .map(|s| runtimepb::HostedService { name: s.clone() })
+            .collect();
+        deployment.hosted_gateways = self
+            .hosted_gateways
+            .iter()
+            .map(|s| {
+                cfg.infra
+                    .as_ref()
+                    .expect("infra not found in runtime config")
```

Comment:
Do we want `unwrap_or_default()` instead to fall back to the default value if it's missing?